samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+        for (int i = 0; i < config.getThreads(); i++) {
+            QuickPerf thread = new QuickPerf(config);
+            if (config.getSamplingQuery() != null) {
+                thread.runSampling();
+            }
+            thread.start();
+            threadList.add(thread);
+        }
+
+        // ProgressBar Tracker Thread
+        ProgressTracker progressTracker = progressTracker = new ProgressTracker(threadList,
+                config.getIterations() * config.getThreads());
+
+        progressTracker.start();
+
+        progressTracker.join();
+
+        int i = 0;
+        for (QuickPerf thread : threadList) {
+            thread.join();
+
+            if (i == 0) {
+                measures = thread.getMeasures();
+            } else {
+                measures = ArrayUtils.addAll(measures, thread.getMeasures());
+            }
+            i++;
+        }
+
+        // write to file before its sorted
+        if (config.getWriteMetricToFile()) {
+            try {
+                writeMeasuresToFile(measures, MEASURES_FILE_NAME);
+            } catch (IOException e) {
+                System.err.println("An error occurred while writing the file: " + e.getMessage());
+            }
+        }
+
+        System.out.println("\n" + BREAK_STR);
+        System.out.println("Query: " + config.getQuery());
+        System.out.println("Params: " + config.paramsToString());
+        System.out.println("Tag: " + DEFAULT_TAG);
+        if (config.getBatchSize() > 0) {
+            System.out.println("Batching Enabled (size): " + config.getBatchSize());
+        }
+        System.out.println(String.format("Start: %s End: %s", testStartTimestamp, ZonedDateTime.now()));
+        System.out.println(
+                String.format(
+                        "Finished with a total of %s runs across %s Threads.\nLatencies (ms):  p50 = %s, p95 = %s, p99 = %s, min = %s,  max = %s",
+                        config.getIterations() * config.getThreads(), config.getThreads(),
+                        calcPerc(measures, 50), calcPerc(measures, 95), calcPerc(measures, 99), getMin(measures),
+                        getMax(measures)));
+        System.out.println(BREAK_STR);
+
+    }
+
+    public QuickPerf(Config config) {
+        this.config = config;
+    }
+
+    private void runSampling() {
+        System.out.println("Running Sampling... ");
+
+        String connectionUrl = createConnectionURL(config);
+
+        try (Connection connection = DriverManager.getConnection(connectionUrl)) {
+            try (Statement statement = connection.createStatement()) {
+                boolean hasResults = statement.execute(config.getSamplingQuery());
+
+                if (hasResults == false) {
+                    System.out.println("Nothing sampled");
+                    return;
+                }
+
+                ResultSet rs = statement.getResultSet();
+                while (rs.next()) {
+                    String value = rs.getString(1);
```

Comment:
This assumes that the sampling query always returns something that can always be retrieved as a string. It might be safer to define the list as `List<Object>` and call `ResultSet#getObject(int)` instead.

Reply:
right now the sampling query assumes one returns a String - needs to be casted and gets converted into either Int or String for downstream use. This is a limitation that could get extended later.